### PR TITLE
#36 Refactored first screen code

### PR DESCRIPTION
**Pull request by Chopikov**
- Changed IDs in xml
- Changed strings names
- Added comments
- Changed class member ordering

### CODE REVIEW

File: app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java [hunk 1]
```diff
@@ -110,33 +122,19 @@ public void onClick(DialogInterface dialog, int id) {
                         dialog.cancel();
                     }
                 })
-                .setPositiveButton("Change", new DialogInterface.OnClickListener() {
+                .setPositiveButton("Change",
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java]: Please, extract string resource here

File: app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java [hunk 2]
```diff
@@ -110,33 +122,19 @@ public void onClick(DialogInterface dialog, int id) {
...
-        private EditImageActivity activity;
-
-        void link(EditImageActivity act) {
-            activity = act;
-        }
-
-        void unLink() {
-            activity = null;
-        }
+    static class AsyncTaskConvertImage extends AsyncTask<Void, Void, Void> {
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java:148]: Please, write a comment for the inner class

File: app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java [hunk 3]
```diff
@@ -100,6 +92,26 @@ public void onBackPressed() {
...
 
+    @Override
+    public Object onRetainCustomNonConfigurationInstance() {
+        super.onRetainCustomNonConfigurationInstance();
+        if (_atConvert != null) _atConvert.unLink();
+        return _atConvert;
+    }
+
+    /**
+     * Provides cancellation of asynch task execution when the dialog is cancelled
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java]: Commonly used `async`

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
 
-    <string name="edit_image_activity_name">Editing</string>
-    <string name="mi_convert">Convert</string>
-    <string name="btn_crop">Crop</string>
-    <string name="btn_columns">Columns</string>
+    <string name="title_edit_image_activity">Editing</string>
+    <string name="action_convert">Convert</string>
+    <string name="action_open_crop">Crop</string>
+    <string name="action_open_columns">Columns</string>
     <string name="img_edit_desc">Convertible image</string>
```
Comment (maik-nack) [inline app/src/main/res/values/strings.xml]: Please change the name of this string resource

File: app/src/main/res/menu/menu_edit_image.xml [hunk 5]
```diff
@@ -4,7 +4,7 @@
 
     <item
         android:id="@+id/miConvert"
```
Comment (maik-nack) [inline app/src/main/res/menu/menu_edit_image.xml]: Please change the id of this menu item

File: app/src/main/res/layout/activity_edit_image.xml [hunk 6]
```diff
@@ -30,13 +30,13 @@
             android:drawablePadding="5dp"
             android:paddingTop="8dp"
             android:paddingBottom="5dp"
-            android:text="@string/btn_crop"
+            android:text="@string/action_open_crop"
             android:textColor="@color/colorText"
```
Comment (maik-nack) [inline app/src/main/res/layout/activity_edit_image.xml]: Please change names of colour resources or add new

File: app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java [hunk 7]
```diff
@@ -148,20 +146,31 @@ protected Void doInBackground(Void... params) {
...
-            );
+            _activity._pdLoading.dismiss();
+            Intent intent = new Intent(_activity, ResultActivity.class);
+            _activity.startActivity(intent);
+            _activity.overridePendingTransition(R.anim.slide_in_left,
+                    R.anim.slide_out_left);
+        }
+
+        /**
+         * Get link to EditImageActivity
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java:175]: Please make comment for function in Javadoc style (see CodingStandard.md)

**Comment by maik-nack** [comment]
Please change names of colour resources or add new

**Comment by maik-nack** [comment]
Please change the id of the `linear layout` in the `activity_edit_image.xml`

**Comment by maik-nack** [comment]
Please rename `menu_edit_image.xml` to `edit_image.xml` (see CodingStandard.md)

**Comment by maik-nack** [comment]
Strings 46, 47 and 63, 64 can be replaced by the call of one private function

**Review by PawelMelnik — APPROVED** [review]
Looks nice

**Review by maik-nack — APPROVED** [review]
Nice

**Pull request merged**